### DISCUSSION TITLE
How to mark the output of compare command

### DISCUSSION THREAD

**Discussion by sharad-ahire** [ImageMagick/ImageMagick#3075] [Help] [answered]
Hi All,
We are using IM for our visual regression testing tool where we are comparing 2 images and identifying differences between 2 images. if the difference is big it can easily visible to human eyes but when the difference is very minor ~ 10 pixel them it very hard to see. I am looking for something where we can mark minor difference using rectangle shape so that it we can easily spot very small diff.

Following is the command we are using
`magick compare -highlight-color magenta -metric AE -fuzz 3% text_field_baseline.png text_field_checkpoint.png  -compose exclusion text_field_diff.png`

**Compare command output**
![image](https://user-images.githubusercontent.com/1021411/103716811-c550aa80-4f78-11eb-916b-027fdb46cb4e.png)

**I am looking for following output image from compare command**
![image](https://user-images.githubusercontent.com/1021411/103716934-106abd80-4f79-11eb-9bbb-b40456395d89.png)

**Comment by fmw42** [comment, marked as answer]
You can do connected components on the difference image. The bounding box for each separate region is listed and you can then draw a rectangle on the image. See https://imagemagick.org/script/connected-components.php
